src/main/resources/lib/unsplash/client.ts

```diff
@@ -0,0 +1,53 @@
+import { request } from "/lib/http-client";
+import { SearchPhotosParams, UnsplashResponseBody } from "/lib/unsplash/types";
+
+export function searchPhotos({
+  query,
+  count,
+  start,
+}: SearchPhotosParams): UnsplashResponseBody {
+  const unsplashPageNumber = Math.floor(start / count);
+
+  const accessKey = app.config.unsplashAccessKey;
+  const url = "https://api.unsplash.com/search/photos";
+  const response = request({
+    url,
+    headers: {
+      Authorization: `Client-ID ${accessKey}`,
+    },
+    params: {
+      query,
+      page: unsplashPageNumber.toString(),
+      per_page: count.toString(),
+    },
+  });
+
+  if (response.status < 400 && response.body) {
+    return JSON.parse(response.body);
+  } else {
+    throwErrorFromBodyString(response.body);
+  }
+}
+
+export function getPhotoById(id: string) {
+  const accessKey = app.config.unsplashAccessKey;
+  const url = `https://api.unsplash.com/photos/${id}`;
+  const response = request({
+    url,
+    headers: {
+      Authorization: `Client-ID ${accessKey}`,
+    },
+  });
+
+  if (response.status < 400 && response.body) {
+    return JSON.parse(response.body);
+  } else {
+    throwErrorFromBodyString(response.body);
+  }
+}
+
+function throwErrorFromBodyString(body: string | null): never {
```

Comment:
I've not seen `never` returned before. How would this interact with a return type in `getPhotoById()`?

Reply:
I got inspiration from [this](https://stackoverflow.com/questions/49434751/how-to-declare-a-function-that-throws-an-error-in-typescript) answer on StackOverflow.

Reply:
Nice! Now I've learned something new.